### PR TITLE
firefox: updates

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
Upstream release day was yesterday.

NSS 3.52.1 instead of 3.53, because it fails to compile with:
```
In file included from blapi.h:11,
                 from loader.h:11,
                 from loader.c:8:
blapit.h:11:10: fatal error: seccomon.h: No such file or directory
   11 | #include "seccomon.h"
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [../../coreconf/rules.mk:292: Linux5.4_x86_64_gcc_glibc_PTH_64_OPT.OBJ/loader.o] Error 1
make[2]: Leaving directory '/build/nss-3.53/nss/lib/freebl'
make[1]: *** [../coreconf/rules.mk:44: freebl] Error 2
make[1]: Leaving directory '/build/nss-3.53/nss/lib'
make: *** [coreconf/rules.mk:44: lib] Error 2
```

cc @andir

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nss/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ in stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "mirror://mozilla/security/nss/releases/NSS_${underscoreVersion}_RTM/src/${pname}-${version}.tar.gz";
-    sha256 = "0q8m9jf6zgkbhx71myjb7y0gcl5ib3gj6qkl9yvdqpd6vl6fn2ha";
+    sha256 = "sha256-vMgawzrrTsrRgtwh400sl7zBSP17nnb4W+u4kkBakng=";
```
Comment (vcunat) [inline pkgs/development/libraries/nss/default.nix:17]: Please, don't use SRI hashes in official nixpkgs yet.  See #89275.

Comment (ajs124) [reply pkgs/development/libraries/nss/default.nix]: Fixed.

**Comment by andir** [comment]
Usually we have that NSS update as part of a staging/staging-next cycle to reduce the amounts of rebuilds on master. Given the severity of some of the issues and it being a dependency of Firefox it might be fine to just land this on master as is. If someone disagrees we can also do it the "normal" way.

I'll spent the afternoon on the backports of this. As of a few days we are only support 20.03 and master IIRC so I'll skip 19.09 but it would probably not be that hard (famous last words).

**Comment by FRidh** [comment]
I expect tomorrow around noon we can start a new staging-next cycle so this can be included then.

**Comment by ajs124** [comment]
I'm not that familiar with the whole staging/staging-next workflow. Is there anything I need to do once the new cycle has been started?

**Comment by vcunat** [comment]
I pushed the `nss` bump to staging now.  This still does not prevent us from skipping the wait in case the staging-next cycle gets stuck on some hard-to-solve problem.  (I pushed with non-SRI hash, although that change perhaps wasn't really required.)  Actually, this time I see no "critical" security problems, so it seems to be less important than usual :-)

**Comment by vcunat** [comment]
Uh, I forgot to mention it here... and forgot this PR for a while: most of the commits don't need any large rebuild and they can go directly to master.  I pushed those now (better late than later).

**Comment by vcunat** [comment]
I see these commit hashes are linked from 20.03 already, so we merge this PR as-is into current staging-next?

**Comment by ajs124** [comment]
I don't know why @edolstra merged the backport, before this was merged. Since the commits are already referenced on release-20.03, this should probably merged as is.

**Comment by vcunat** [comment]
I think it's good that 20.03 merge to wasn't delayed this long, but we could have merged this whole PR to staging long ago &ndash; at the first moment I only took what was necessary (`nss`).

**Comment by ajs124** [comment]
So I'll just close this then, right?

**Comment by FRidh** [comment]
Won't firefox on master fail because nss is still on staging-next?

**Comment by vcunat** [comment]
`firefox` would fail, but I didn't push that.  I tested running all those I pushed.

**Pull request merged**